### PR TITLE
[t] use one base class

### PR DESCRIPTION
**Pull request by mikz**
unify all tests to use one base
so we can share setting between tests

### CODE REVIEW

File: apicast/src/configuration.lua [hunk 1]
```diff
@@ -129,12 +128,34 @@ end
...
 local Service = require 'configuration.service'
 
+local noop = function() end
+local function readonly_table(table)
+    return setmetatable({}, { __newindex = noop, __index = table })
+end
+
+local empty_t = readonly_table()
+
+local function backend_endpoint(proxy)
```
Comment (davidor) [inline apicast/src/configuration.lua]: The indentation here is not consistent with the rest of the codebase.

Comment (mikz) [reply apicast/src/configuration.lua]: Indeed. Haven't configured my IDE and used the default 4 spaces indentation. 
Must admit that I quite like it. Somehow it makes it more readable. Also the OpenResty code is indented with 4 spaces. 
I'd consider switching to 4 spaces. Wdyt?

Comment (davidor) [reply apicast/src/configuration.lua]: I don't mind as long as it's spaces :) and we are consistent across the codebase.
http://lua-users.org/wiki/LuaStyleGuide

File: t/001-management.t [hunk 2]
```diff
@@ -1,17 +1,7 @@
-use Test::Nginx::Socket::Lua 'no_plan';
-use Cwd qw(cwd);
+use lib 't';
 
-my $pwd = cwd();
-my $apicast = $ENV{TEST_NGINX_APICAST_PATH} || "$pwd/apicast";
+use TestAPIcast 'no_plan';
```
Comment (davidor) [inline t/001-management.t]: What is `no_plan` ?

File: t/016-balancer.t [hunk 3]
```diff
@@ -1,22 +1,14 @@
...
+$ENV{TEST_NGINX_HTTP_CONFIG} = "$TestAPIcast::path/http.d/*.conf";
 $ENV{RESOLVER} = '127.0.1.1:5353';
 
-
 env_to_nginx(
     'RESOLVER'
 );
 master_on();
-log_level('debug');
 repeat_each(1);
```
Comment (davidor) [inline t/016-balancer.t]: Should we define `TEST_NGINX_REPEAT_EACH` instead?
From TestAPIcast: `repeat_each($ENV{TEST_NGINX_REPEAT_EACH} || 2);`

File: t/001-management.t (hunk 2 above)
Comment (mikz) [reply t/001-management.t]: I don't really know. I guess it defines in what order to execute the tests. Search for `plan` on http://search.cpan.org/~agent/Test-Nginx/lib/Test/Nginx/Socket.pm

https://openresty.gitbooks.io/programming-openresty/content/ also have few mentions of it.

File: t/016-balancer.t (hunk 3 above)
Comment (mikz) [reply t/016-balancer.t]: Some tests force one iteration because 2nd one would hit cache and have different result.

Because the repeat_each is in the base this is an override. So all tests default to 2 (or the env var) except few ones that are forced to one.

File: t/TestAPIcast.pm [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package TestAPIcast;
```
Comment (davidor) [inline t/TestAPIcast.pm]: Nice DRY :+1:

File: .circleci/config.yml [hunk 5]
```diff
@@ -45,6 +45,7 @@ jobs:
     working_directory: /opt/app-root/apicast
     steps:
       - checkout
+      - run: rm -rf lua_modules
```
Comment (davidor) [inline .circleci/config.yml]: Why is this needed now, but not before? :thinking:

Comment (mikz) [reply .circleci/config.yml]: It is needed for local run. There is no way to ignore folders on local build.
